PaddleAudio/examples/dcase2021_task1b/README.md

```diff
@@ -0,0 +1,117 @@
+- asserts/: all the necessary files/data/config
+
+
+## Dataset
+
+Data was recorded in the following 10 scenes:
+- Airport - airport
+- Indoor shopping mall - shopping_mall
+- Metro station - metro_station
+- Pedestrian street - street_pedestrian
+- Public square - public_square
+- Street with medium level of traffic - street_traffic
+- Travelling by a tram - tram
+- Travelling by a bus - bus
+- Travelling by an underground metro - metro
+- Urban park - park
+
+### Audio-only
+
+For audio-only scene classification, we extract the same feature as described in  [audioset training example](../audioset_training) and use the same network, namely resent50, except that we add extra fc-layers and one final classification head for the 10-way classification. We also add dropout to reduce over-fitting. See [model.py](model.py) for details.
+
+Training audio-only network:
+``` sh
+python train.py --device=0 --task_type=audio_only --config='./assets/config.yaml'
+```
+
+To evaluate using our pre-trained weights, try
+``` sh
+python evaluate.py --device=0 --task_type=audio_only --config='./assets/config.yaml'
+```
+
+
+
+### Video-only
+For video-only task, we use OpenAI CLIP for
+#### Convert video to images
+We first extract images at 1 frame per second from the provided video files for further process. Using ffmpeg, this can be done as follows,
+
+``` sh
+mkdir -p images
+for f in `ls video/*.mp4`; do
+    file=`echo $f |cut -d . -f 1|cut -d / -f 2`
+    ffmpeg -i $f -vf fps=1 ./images/$file-%02d.png
+done
+```
+The above script assumes that you put all video files under folder './video'.
+
+#### Zero-shot classification
+After converting video to images, we use OpenAI clip to extract class probability in zero-shot manner. We use the 5th frame to represent the whole video. The usage of zero-shot image classification is explained in [OpenAI's implementation](https://github.com/openai/CLIP) in Pytorch, and also in a third-party  [implementation](https://github.com/ranchlai/CLIP.paddle).
+
+The overall zero-shot video-only accuracy is 0.697 and 0.729 for validation split and whole dataset respectively without any heavy prompt engineering.
+To evaluate zero-shot accuracy, run
+``` sh
+python evaluate_visual_zs.py
+```
+The above script will automatically download pre-extracted zero-shot features and perform the evaluation.
+
+#### linear-probe classification
+For linear-probe, see [README_LP.md](README_LP.md)
+
+### Audio-visual task
+We can combine audio and video branch in a late-fusion manner. THe network architecture is as follows,
+![network](./assets/network_arch.png)
+
+The audio-visual task utilizes embedding output form CLIP image encoder. For the convenience of this demo, the embedding has been pre-extracted and will be automatically downloaded before training.
+
+
+To train audio-visual network, run
+``` sh
+python train.py --device=0 --task_type='audio_visual' --config='./assets/config.yaml'
```

Comment:
--device=gpu
--device=cpu
--device=xpu (for ASIC chips)
Please update all the --device usage to correct one.